jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/messages/AbstractMessageSink.java

```diff
@@ -18,14 +18,82 @@
 
 import org.eclipse.jetty.websocket.core.CoreSession;
 
+/**
+ * <p>Partial implementation of {@link MessageSink}.</p>
+ * <p>The application function passed to the constructor as a {@link MethodHandle}
+ * either delegates the demand for more WebSocket frames to the {@link MessageSink}
+ * implementation -- and therefore {@code autoDemand==false}, or explicitly demands
+ * for more WebSocket frames (via APIs that eventually call
+ * {@link CoreSession#demand(long)}) -- and therefore {@code autoDemand==true}.</p>
+ * <p>{@link MessageSink} implementations must handle the demand for WebSocket
+ * frames in this way:</p>
```

Comment:
It is not the "application function" that manages demand, but the "application".  The "function" might not even be invoked before the initial demand is required to be issued by the application.

```suggestion
 * <p>Management of demand for message frames may either be entirely managed
 * by the {@link MessageSink} implementation ({@code autoDemand==true}; or
 * it may be managed collaboratively between the application and the 
 * {@link MessageSink} implementation ({@code autoDemand==true}.
 * <p>{@link MessageSink} implementations must handle the demand for WebSocket
 * frames in this way:</p>
```

Reply:
> It is not the "application function" that manages demand, but the "application". 

It is that particular application function that manages demand, not some undefined "application".
Also, the demand is not for "message frames" (they don't exist), but for generic WebSocket frames (including CLOSE).

I have reworded into a mix of the two.